soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -12117,7 +12117,8 @@ void func_8084AEEC(Player* this, f32* arg1, f32 arg2, s16 arg3) {
                 swimMod *= CVarGetFloat(CVAR_SETTING("WalkModifier.SwimMapping2"), 1.0f);
             }
         } else {
```

Comment:
```suggestion
        // sControlInput is NULL to prevent inputs while surfacing after obtaining an underwater item so we want to ignore it for that case
        } else if (sControlInput != NULL) {
```